File: CarProject/API/Entities/Car.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace API.Entities
{
    public class Car
    {
        public Car(Guid id, string make, string model, int year, Guid detailId)
        {
            Id = id;
            Make = make;
            Model = model;
            Year = year;
            DetailId = detailId;

        }
        
        public Guid Id { get; set; }

        [StringLength(60)]
        public string Make { get; set; }

        [StringLength(60)]
        public string Model { get; set; }

        public int Year { get; set; }

        public Guid DetailId { get; set; }

        public virtual Detail Detail { get; set; }

        public virtual List<CarImage> CarImages { get; set;}
    }
}

File: frontend/src/Models/car.ts
import { CarImage } from "./carImage";
import { InteriorDetail } from './InteriorDetail';
import { ExteriorDetail } from './exteriorDetail';
import { EngineDetail } from "./EngineDetail";
import { SafetyDetail } from './safetyDetail';

export interface Car {
    Id: string
    Make: string
    Model: string
    Year: Number
    Images?: Array<CarImage>
    InteriorDetail?: InteriorDetail
    ExteriorDetail?: ExteriorDetail
    EngineDetail?: EngineDetail
    SafetyDetail?: SafetyDetail
}